### PR TITLE
[7/n] [dropshot_endpoint] add tests for custom and unnamed params

### PR DESCRIPTION
**Pull request by sunshowers**
Noticed that channels weren't using "_dropshot_crate" sometimes, so address
that.

As part of the test, introduce a syn visitor that bans particular identifiers.
We're going to use this for trait-based servers as well.

### CODE REVIEW

File: dropshot_endpoint/src/test_util.rs [hunk 1]
```diff
@@ -0,0 +1,44 @@
+// Copyright 2023 Oxide Computer Company
```
Comment (ahl) [inline dropshot_endpoint/src/test_util.rs:1]: 2024

File: dropshot_endpoint/src/test_util.rs [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+            BanIdentsVisitor { idents: &idents, found: BTreeSet::new() };
+        visitor.visit_file(file);
+
+        if !visitor.found.is_empty() {
+            let found = visitor.found.into_iter().collect::<Vec<_>>();
+            panic!("banned identifiers found in file: {}", found.join(", "));
+        }
+    }
+
+    assert_banned_idents(file, idents);
```
Comment (ahl) [inline dropshot_endpoint/src/test_util.rs:25]: what's the point of this inner function?

**Review by ahl — APPROVED** [review]
Would it make sense to have a test that fails the `assert_banned_idents` check? (apologies if I missed it)

File: dropshot_endpoint/src/test_util.rs (hunk 2 above)
Comment (sunshowers) [reply dropshot_endpoint/src/test_util.rs:25]: Avoiding excessive monomorphization on the IntoIterator -- but it's really not necessary, it's a legacy from my first attempt where I didn't just use a visitor and the function was longer. Will fix.

**Comment by sunshowers** [comment]
> Would it make sense to have a test that fails the assert_banned_idents check? (apologies if I missed it)

Good idea -- no I haven't added this yet.

File: dropshot_endpoint/src/test_util.rs (hunk 2 above)
Comment (sunshowers) [reply dropshot_endpoint/src/test_util.rs]: Fixed.

**Comment by sunshowers** [comment]
> Good idea -- no I haven't added this yet.

Done.

**Review by ahl — APPROVED** [review]

**Comment by sunshowers** [comment]
Also added a test for unnamed params here, since we're going to fix this in an upcoming PR.

**Comment by sunshowers** [comment]
Accidentally broke main 😅 fixing it in this commit because it's pretty trivial.

**Pull request merged**